src/runtime_src/core/common/query_requests.h

```diff
@@ -17,6 +17,7 @@
 #ifndef xrt_core_common_query_requests_h
 #define xrt_core_common_query_requests_h
 
+#include "core/include/xclerr_int.h"
```

Comment:
Let's not include this one here.  You should move the parsing routine implementation to `query_requests.cpp` where this header is then included.  I do see that you need a forward declaration of `struct xclErrorLast` for your parsing function declarations.

Reply:
Difficult. It has enum xrtErrorClass as well.